types/validator_set_test.go

```diff
@@ -62,8 +62,12 @@ func TestValidatorSetBasic(t *testing.T) {
 
 	// update
 	assert.False(t, vset.Update(randValidator_()))
-	val.VotingPower = 100
+	_, val = vset.GetByAddress(val.Address)
+	val.VotingPower += 100
+	proposerPriority := val.ProposerPriority
```

Comment:
We still need proper tests as suggested in #2942 but this change would test the changed behaviour:
```suggestion
	proposerPriority := val.ProposerPriority
	// Mimic update from types.PB2TM.ValidatorUpdates which does not know about ProposerPriority
	// and hence chooses the default value 0.
	val.ProposerPriority = 0
```